### PR TITLE
panels-graph start point

### PR DESCRIPTION
**Pull request by Vane11ope**
implemented calling graph as one of panels very roughly. more sophistication is definitely coming super soon. this is just an entry point of that.

1. must come up with the better drawing algorithm
2. resizing does not work yet
3. a pile of things i have in mind that i just can't write all down yet.

btw to activate the mode, go to V! -> G

### CODE REVIEW

**Comment by radare** [comment]
all graph tests fail :D

**Comment by Vane11ope** [comment]
what could i do about it? :'O

**Comment by ret2libc** [comment]
I did `V!G` but I just saw the normal graph.

BTW, why do you need to know, in graph.c, if you are in panels or not? I think that's the wrong place to put things.

**Comment by Vane11ope** [comment]
@ret2libc i need to know if this is from panels or not because panel needs to call r_core_visual_graph to draw graph yet it does not come back to panels once the graph API is called. 
so i had to let the graph take care of all the drawings and to do that, i need to let the graph know if this is from panels or not so that the graph can see if it needs to call some panels API to draw panels on it.

**Comment by Vane11ope** [comment]
now been only a line added to graph

File: libr/core/panels.c [hunk 1]
```diff
@@ -533,6 +556,18 @@ static void zoom(RPanels *panels) {
 	}
 }
 
+static bool checkFunc(RCore *core) {
+	RAnalFunction *fun = r_anal_get_fcn_in (core->anal, core->offset, R_ANAL_FCN_TYPE_NULL);
+	if (!fun) {
+		r_cons_message ("Not in a function. Type 'df' to define it here");
+		return false;
+	} else if (r_list_empty (fun->bbs)) {
```
Comment (radare) [inline libr/core/panels.c]: dont else after return

Comment (Vane11ope) [reply libr/core/panels.c]: right. fixed.

**Comment by ret2libc** [comment]
Though now it's much better, I think the approach is wrong.
If you see at other panels contents, I think they are not aware of being in a panel. The same should be true for graph. For example, with this PR even if you are in panel mode, all keybindings of graph are active, but those of the panel are not.

Have you tried to draw the graph in panels with `r_core_visual_graph(..., ..., ..., false);`? This should work _I think_.

**Comment by Vane11ope** [comment]
thank you and yeah im fully aware of key bindings. i was gonna fix those in next PR. i will check the way of calling graph you taught. i actually didn't know about that :(

**Pull request merged**